wweval/R/ms_fig4.R

```diff
@@ -195,53 +224,153 @@ make_fig4_admissions_overall <- function(eval_hosp_data,
   return(p)
 }
 
+#' Get the mean relative crps for a location
+#'
+#' @param scores tibble of scores by day forecast day model
+#' @param locs loc to get mean relative score for
+#'
+#' @return table of mean relative score for each location
+get_loc_rel_crps <- function(scores, locs) {
+  relative_crps <- scores |>
+    dplyr::filter(location %in% locs) |>
+    dplyr::group_by(location, model) |>
+    dplyr::summarise(mean_crps = mean(crps)) |>
+    tidyr::pivot_wider(
+      names_from = model,
+      values_from = mean_crps
+    ) |>
+    dplyr::mutate(rel_mean_crps = ww / hosp)
+
+  return(relative_crps)
+}
+
+#' Plot a heatmap of the relative crps by locations and forecast date
+#' for the head-to-head comparison
+#'
+#' @param scores A tibble of daily scores by forecast date, location, and model
+#' @param fig_file_dir A string indicating the directory to save the figures in
+#'
+#' @return a ggplot object
+#' @export
+get_plot_rel_crps_heatmap <- function(scores,
+                                      fig_file_dir) {
+  scores_summary <- scores |>
+    compute_relative_crps(id_cols = c(
+      "location",
+      "forecast_date", "date"
+    )) |>
+    dplyr::group_by(location, forecast_date) |>
+    dplyr::summarize(
+      mean_rel_crps = mean(rel_crps)
+    )
```

Comment:
this looks like it's being used in the figure. I thought the plan was to use `rel_mean` throughout.

Reply:
I think this is a mistake, will address tomorrow. I think I thought I changed it in everything but missed this

Reply:
Now fixed.